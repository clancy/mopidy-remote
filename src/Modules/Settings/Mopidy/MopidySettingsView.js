import React, { Component, PropTypes } from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
  TouchableOpacity
} from 'react-native';
import FormRow from './FormRow'
import * as MopidyActions from '../../Mopidy/Actions'

export default class MopidySettingsView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hostname: props.hostname,
      port: props.port
    };
  };
  static propTypes = {
    hostname: PropTypes.string,
    port: PropTypes.string
  };
  connect = () => {
    var hostname = this.state.hostname;
    var port = this.state.port;
    //TODO validate hostname and port
    this.props.dispatch(MopidyActions.connect(hostname, port));
  };
  render() {
    return (
      <View style={styles.container}>
        <FormRow
          label="Host Name"
          value={this.state.hostname}
          onChangeText={(hostname) => this.setState({hostname})} />
        <FormRow
          label="Port"
          value={this.state.port}
          onChangeText={(port) => this.setState({port})} />
        <View style={styles.buttonRow}>
          <TouchableOpacity
            style={styles.button}
            onPress={this.connect}>
            <Text style={styles.buttonText}>CONNECT</Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'black',
    flex: 1
  },
  buttonRow: {
    height: 40,
    flexDirection: 'column',
    alignItems: 'center',
    margin: 5
  },
  buttonText: {
    color: 'black',
    fontWeight: 'bold'
  },
  button: {
    justifyContent: 'center',
    backgroundColor: 'white',
    paddingLeft: 40,
    paddingRight: 40,
    borderRadius: 20,
    flex: .3
  },
});
