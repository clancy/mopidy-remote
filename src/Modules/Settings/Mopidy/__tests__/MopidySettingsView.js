import 'react-native';
import React from 'react';
import MopidySettingsView from '../MopidySettingsView';
import { shallow, mount, render } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
describe('MopidySettingsView tests', () => {
  test('renders MopidySettingsView correctly', () => {
    const tree = renderer.create(
      <MopidySettingsView
        hostname={"192.168.1.1"}
        port={"6680"}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('MopidySettingsView connect calls connect function', () => {
    var controlMock = jest.fn();
    const tree = shallow(
      <MopidySettingsView
        hostname={"192.168.1.1"}
        port={"6680"}
        connect={(h, p) => { controlMock(h, p)}}
        />
    );

    tree.find('TouchableOpacity').simulate('press');
    
    expect(controlMock.mock.calls.length).toBe(1);
    expect(controlMock.mock.calls[0][0]).toBe('192.168.1.1');
    expect(controlMock.mock.calls[0][1]).toBe('6680');
  });
});
