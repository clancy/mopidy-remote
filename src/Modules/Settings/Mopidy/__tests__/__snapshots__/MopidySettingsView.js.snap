// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MopidySettingsView tests renders MopidySettingsView correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 40,
        "margin": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "width": 80,
        }
      }
    >
      Host Name
    </Text>
    <TextInput
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "white",
          "borderWidth": 1,
          "color": "white",
          "flex": 1,
        }
      }
      value="192.168.1.1"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 40,
        "margin": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "width": 80,
        }
      }
    >
      Port
    </Text>
    <TextInput
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "white",
          "borderWidth": 1,
          "color": "white",
          "flex": 1,
        }
      }
      value="6680"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 40,
        "margin": 5,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 20,
          "flex": 0.3,
          "justifyContent": "center",
          "opacity": 1,
          "paddingLeft": 40,
          "paddingRight": 40,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontWeight": "bold",
          }
        }
      >
        CONNECT
      </Text>
    </View>
  </View>
</View>
`;
